### PR TITLE
I changed the void loop by adding the button state. When the button i…

### PR DESCRIPTION
**Pull request by tungnguyen1234**
…s on, the code will allows the hand to execute the code. If not, then the button goes into sleep mode.

### CODE REVIEW

File: myoFlex/myoFlex.ino [hunk 1]
```diff
@@ -39,22 +40,42 @@ Serial.begin(9600);     //starts up serial communication between arduino and com
   pinky.attach(pinkyPin, min, max);
   Serial.println("Begin");    //prints a start message
   handPosition(max,max,max,max,max);       //makes the hand relax all of the way
+  // Initialize the pushbutton pin as an input:
+  pinMode(buttonPin, INPUT);
```
Comment (jdpeiffer6) [inline myoFlex/myoFlex.ino]: great!

File: myoFlex/myoFlex.ino [hunk 2]
```diff
@@ -39,22 +40,42 @@ Serial.begin(9600);     //starts up serial communication between arduino and com
...
+  #Reduce power consumption: 
+  #One way:turn off from analog to digital 
+  ADCSRA &= ~(1<<7); #\~ makes 0 bits to 1 \and 1 bits to 0 
+  
+  #Enable sleep mode
+  SMCR |= (1<<2); # Set the SM2 by setting 1 \and \move to the 
+  #corresponding binary code
+  SMCR |= 1; # Set the SE
+  #Watchdog timer
+  __asm__  __volatile__("sleep");//in line assembler to go to sleep
```
Comment (jdpeiffer6) [inline myoFlex/myoFlex.ino]: My only question is does this require any additional libraries?

File: myoFlex/myoFlex.ino [hunk 3]
```diff
@@ -39,22 +40,42 @@ Serial.begin(9600);     //starts up serial communication between arduino and com
...
 // THIS PART OF THE CODE LOOPS CONTINIOUSLY 
 void loop() {
- currentVoltage = analogRead(myoIn);     //reads myoelectric input
- currentVoltage=currentVoltage/1023;      // converts from 0-1023 scale to 0-1 scale
- Serial.println(currentVoltage);        //puts it in the serial moniter
- if (contract==false && currentVoltage>threshold){      //if we see a spike and the hand is relaxed
- Serial.println("Contract!");     
+  buttonState = digitalRead(buttonPin);
+  #REduce power consumption.
+  if (buttonState == HIGH){
```
Comment (jdpeiffer6) [inline myoFlex/myoFlex.ino]: Do you think using the LOW button state would be better here? At least for prototyping so we don't have to plug in the button every time if were just testing?

File: myoFlex/myoFlex.ino (hunk 2 above)
Comment (tungnguyen1234) [reply myoFlex/myoFlex.ino]: As I checked, the author didn't use any further libraries. One thing is that the constant buttonPin needed to adjust according to our Arduino Pro.

**Pull request merged**